### ISSUE TITLE
Multiple 'it' in one 'describe' with a beforeEach

### ISSUE THREAD

**Issue by nataliejuner** [cypress-io/cypress#4171]
## Current behavior:
Currently, I am running a ```beforeEach``` and it resets at the beginning of each test which is what I want but also no. I have tests that are independent but grouped together because they are testing the same component. So, for now it looks like this: 
```
describe('this is a test',function(){
	it('will be a test example',function(){
		cy.contains('test').click
		cy.contains('another test').click()
	})
})
```

And my beforeEach looks like this: 
```
beforeEach(() => {
  cy.visit('http://localhost:3000/');
});
```

## Desired behavior:
I want to be able to write my tests like this: 
```
describe('this is a test',function(){
	it('will be a test example',function(){
		cy.contains('test').click
	})
	it('will test another test',function(){
		cy.contains('another test').click()
	})
})
```

## Summary
So is there anyway to not have the beforeEach effect that second ```it``` in my desired behavior? or will I have to write a cy.visit() in each test?

**Comment by kylemh** [comment]
You _can_ have multiple `it` within the scope of a single `describe`. You can honestly keep nesting layers of `describe` and `it` as freely as you'd like, as long as there are no `describe` nested within an `it`.

The real issue seems to be your summary...

Your `beforeEach` will run before every single it and describe within it's own describe's scope.

If you have a test that you dont want affected by a `describe`'s `beforeEach`, consider making a new describe, outside of the other's scope... like so:

```javascript
describe('test suite', () => {
  beforeEach(() => {});

  it('test 1', () => {});

  it('test 2', () => {});
});

describe('test suite 2', () => {
  beforeEach(() => {});

  it('test 1', () => {});

  it('test 2', () => {});
});
```

**Comment by nataliejuner** [comment]
So my beforeEach is in the index.js>Support>Cypress

Is that why it's resetting every time because it's not written the way you wrote in your example?

**Comment by kylemh** [comment]
If you have a `beforeEach` inside `cypress/support/index.js`, that `beforeEach` will get called on every single one of your tests, in every suite, in every integration file.

https://docs.cypress.io/guides/references/best-practices.html#State-reset-should-go-before-each-test

**Comment by nataliejuner** [comment]
I gotcha. Thanks for the answer; I will most likely just re-write the code to include the ```beforeEach``` in every test EXCEPT the one and just use the regular ```cy.visit()``` so that I can have multiple ```it``` functions.

**Comment by Shakhadat** [comment]
I have the same error I could not solve it, 
-----------
Timed out after waiting 60000ms for your remote page to load.



Your page did not fire its load event within 60000ms.



You can try increasing the pageLoadTimeout value in cypress.config.js to wait longer.



Browsers will not fire the load event until all stylesheets and scripts are done downloading.



When this load event occurs, Cypress will continue running commands.



Because this error occurred during a before all hook we are skipping all of the remaining tests.
------------------------
 
How can I write secont test case, beforeEach, before working only once the next time could not find CY.visit(...) how can I manage this issue????

**Comment by nagash77** [comment]
@Shakhadat,  If you are seeing this behavior in current versions of Cypress please open a new ticket with a reproducible example and we will be happy to investigate.

Forking [Cypress Test Tiny](https://github.com/cypress-io/cypress-test-tiny) makes sharing a reproducible example easier to share and easier for our engineers to replicate your issues. This method also keeps the reproduction as simple as possible, which helps us eliminate potential causes and noise from the investigation.
